src/features/offer/components/OfferContent/OfferContentBase.tsx

```diff
@@ -0,0 +1,139 @@
+  children: (body: ReactElement) => ReactElement
+  onOfferPreviewPress: (index?: number) => void
+  footer?: ReactElement | null
+  contentContainerStyle?: StyleProp<ViewStyle>
+}
+
+const DELAY_BEFORE_CONSIDERING_PAGE_SEEN = 5000
+
+export const OfferContentBase: FunctionComponent<OfferContentBaseProps> = ({
+  offer,
+  searchGroupList,
+  subcategory,
+  showOfferPreview,
+  footer,
+  children,
+  onOfferPreviewPress,
+  contentContainerStyle,
+}) => {
+  const { userLocation } = useLocation()
+  const {
+    sameArtistPlaylist,
+    sameCategorySimilarOffers,
+    apiRecoParamsSameCategory,
+    otherCategoriesSimilarOffers,
+    apiRecoParamsOtherCategories,
+  } = useOfferPlaylist({ offer, offerSearchGroup: subcategory.searchGroupName, searchGroupList })
+
+  const logConsultWholeOffer = useFunctionOnce(() => {
+    analytics.logConsultWholeOffer(offer.id)
+  })
+
+  const { shouldTriggerBatchSurveyEvent, trackBatchEvent, trackEventHasSeenOfferOnce } =
+    useOfferBatchTracking({
+      offerNativeCategory: subcategory.nativeCategoryId,
+    })
+
+  const offerImages = useMemo(
+    () => (offer.images ? getOfferImageUrls(offer.images) : []),
+    [offer.images]
+  )
+
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout
+    if (shouldTriggerBatchSurveyEvent) {
+      timeoutId = setTimeout(() => {
+        trackBatchEvent()
+      }, DELAY_BEFORE_CONSIDERING_PAGE_SEEN)
+    }
+
+    return () => clearTimeout(timeoutId)
+  }, [shouldTriggerBatchSurveyEvent, trackBatchEvent])
+
+  const scrollEventListener = useCallback(
+    ({ nativeEvent }: NativeSyntheticEvent<NativeScrollEvent>) => {
+      if (isCloseToBottom(nativeEvent)) {
+        logConsultWholeOffer()
+        if (shouldTriggerBatchSurveyEvent) {
+          trackBatchEvent()
+        }
+      }
+    },
+    [logConsultWholeOffer, shouldTriggerBatchSurveyEvent, trackBatchEvent]
+  )
+
+  const { headerTransition, onScroll } = useOpacityTransition({
+    listener: scrollEventListener,
+  })
+
+  return (
+    <Container>
+      <OfferWebMetaHeader offer={offer} />
+      <OfferHeader title={offer.name} headerTransition={headerTransition} offer={offer} />
+      <ScrollViewContainer
+        testID="offerv2-container"
+        scrollEventThrottle={16}
+        scrollIndicatorInsets={scrollIndicatorInsets}
+        bounces={false}
+        contentContainerStyle={contentContainerStyle}
+        onScroll={onScroll}>
+        {children(
+          <React.Fragment>
+            <OfferImageContainer
+              imageUrls={offerImages}
+              categoryId={subcategory.categoryId}
+              shouldDisplayOfferPreview={showOfferPreview}
+              onPress={offerImages.length > 0 ? onOfferPreviewPress : undefined}
+            />
+            <OfferBody
+              offer={offer}
+              subcategory={subcategory}
+              trackEventHasSeenOfferOnce={trackEventHasSeenOfferOnce}
+            />
+          </React.Fragment>
+        )}
```

Comment:
je ne sais pas si c'est une bonne idée, si on passe l'argument à children dans un objet, ça permettrait de passer un composant sans passer par une fonction

```suggestion
        {children({ children: (
          <React.Fragment>
            <OfferImageContainer
              imageUrls={offerImages}
              categoryId={subcategory.categoryId}
              shouldDisplayOfferPreview={showOfferPreview}
              onPress={offerImages.length > 0 ? onOfferPreviewPress : undefined}
            />
            <OfferBody
              offer={offer}
              subcategory={subcategory}
              trackEventHasSeenOfferOnce={trackEventHasSeenOfferOnce}
            />
          </React.Fragment>
        )})}
```

voir commentaire suivant

Reply:
J'ai finalement retiré la render prop pour children et j'ai ajouté une prop `BodyWrapper` qui est de type `FunctionComponent`.